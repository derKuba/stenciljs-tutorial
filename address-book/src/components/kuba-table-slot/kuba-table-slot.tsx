import { Component, Host, h } from '@stencil/core';

@Component({
  tag: 'kuba-table-slot',
  styleUrl: 'kuba-table-slot.css',
  shadow: true,
})
export class KubaTableSlot {

  render() {
    return (
      <Host>
        <table>
          <thead>
            <slot name="table-head" />
          </thead>
          <tbody>
            <slot />
          </tbody>
        </table>
      </Host>
    );
  }

}
