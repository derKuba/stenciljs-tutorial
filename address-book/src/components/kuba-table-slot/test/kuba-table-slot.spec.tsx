import { newSpecPage } from "@stencil/core/testing";
import { KubaTableSlot } from "../kuba-table-slot";

describe("kuba-table-slot", () => {
  it("renders", async () => {
    const page = await newSpecPage({
      components: [KubaTableSlot],
      html: `<kuba-table-slot>
      <tr slot="table-head">
          <th>id</th>
          <th>Name</th>
          <th>Vorname</th>
      </tr>
      <tr>
          <td>1</td>
          <td>Max</td>
          <td>Muster</td>
      </tr>
      <tr>
          <td>2</td>
          <td>Maxine</td>
          <td>Muster</td>
      </tr>
      <tr>
          <td>3</td>
          <td>Momo</td>
          <td>Muster</td>
      </tr>
  </kuba-table-slot>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
