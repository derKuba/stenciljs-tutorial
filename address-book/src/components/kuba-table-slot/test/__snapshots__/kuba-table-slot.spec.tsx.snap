// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kuba-table-slot renders 1`] = `
<kuba-table-slot>
  <mock:shadow-root>
    <table>
      <thead>
        <slot name="table-head"></slot>
      </thead>
      <tbody>
        <slot></slot>
      </tbody>
    </table>
  </mock:shadow-root>
  id Name Vorname 1 Max Muster 2 Maxine Muster 3 Momo Muster
</kuba-table-slot>
`;
